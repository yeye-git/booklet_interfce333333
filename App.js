import * as React from 'react';
import { Button, Text, View, Image, Alert } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import SignUp from './screens/signup';
import Setting from './screens/settings';
import Login from './screens/login';
import CommonBar from './common-file/NavigationBar';
import Teacher_student_HomePage from './common-file/Teacher_student_homepage';
import TeacherProfile from './screens/teacher/Profile-Teacher';
import create from 'zustand';
import BookSelf from './common-file/Teacher_Student_BookshelfPage';
import ForgetPassword from './common-file/Forgrt_Password';
import CreateNewBookletPage from './screens/teacher/CreateNewBookletPage';
import TeacherStudentMessage from './common-file/Teacher_Student_Message';
import TeacherStudentConversation from './common-file/Teacher_Student_Conversation';
import { ClassFrame } from './screens/teacher/Class';
import DemoBooklet from './screens/student/Booklet';
function LogoTitle1(props) {
    return (
        <View>
            <View style={{ backgroundColor: 'purple', width: '100%' }}>
                <Text>Message</Text>
            </View>
        </View>
    );
}

function LogoTitle2() {
    return (
        <View>
            <Text>7890</Text>
        </View>
    );
}

const Stack = createStackNavigator();

//main func entry
export default function App() {
    const useStore = create((set) => ({
        roleVariable: 0,
        userName: 'JoJo',
        count: 0,
        inc: () => set((state) => ({ count: state.count + 1 })),
        setRoleVariable: (val) => set((state) => ({ roleVariable: val })),
        setUserName: (val) => set((state) => ({ userName: val })),
        setUser: (val) => set((state) => ({ user: val })),
    }));

    const Bar = (props) => {
        console.log(props, 'pppp');

        return (
            <View>
                <Text>My profile</Text>
            </View>
        );
    };

    function HomeScreen2({ navigation }) {
        const [count, setCount] = React.useState(0);

        React.useLayoutEffect(() => {
            navigation.setOptions({
                headerRight: () => <Button onPress={() => setCount((c) => c + 1)} title="Update count" />,
            });
        }, [navigation, setCount]);

        return (
            <View style={{}}>
                <StudentCom></StudentCom>
            </View>
        );
    }

    const NewCommonBar = ({ navigation }) => {
        const MessageBar = () => {
            return (
                <View>
                    <TeacherStudentMessage navigation={navigation} />
                </View>
            );
        };

        const BookSelfBar = () => {
            const roleVariable = useStore((state) => state.roleVariable);
            return <BookSelf navigation={navigation} roleVariable={roleVariable}></BookSelf>;
        };

        const ProfileBar = (props) => {
            const roleVariable = useStore((state) => state.roleVariable);
            const useStore_userName = useStore((state) => state.userName);
            const count = useStore((state) => state.count);
            return (
                <View>
                    {/* <Text>111</Text> */}
                    {roleVariable == 'roleStudent' && (
                        // <StudentProfile count={count} useStore_userName={useStore_userName}></StudentProfile>
                        <TeacherProfile
                            navigation={props.navigation}
                            count={count}
                            useStore_userName={useStore_userName}
                        ></TeacherProfile>
                    )}
                    {roleVariable == 'roleTeacher' && (
                        <TeacherProfile
                            navigation={props.navigation}
                            count={count}
                            useStore_userName={useStore_userName}
                        ></TeacherProfile>
                    )}
                </View>
            );
        };

        const bookselfParams = create(navigation.getState().routes.filter((item, index) => item.name === 'HomeName'));

        return (
            <CommonBar
                bookself={<BookSelfBar navigation={navigation}></BookSelfBar>}
                home={(props) => {
                    return (
                        <Teacher_student_HomePage
                            {...props}
                            navigation={navigation}
                            role={useStore((state) => state.roleVariable)}
                        ></Teacher_student_HomePage>
                    );
                }}
                // home={
                //     <Teacher_student_HomePage
                //         navigation={navigation}
                //         role={useStore((state) => state.roleVariable)}
                //     ></Teacher_student_HomePage>
                // }
                message={<MessageBar></MessageBar>}
                profile={<ProfileBar navigation={navigation}></ProfileBar>}
            ></CommonBar>
        );
    };

    const LoginPage = ({ navigation }) => {
        const inc = useStore((state) => state.inc);
        const set_UserName = useStore((state) => state.setUserName);
        // change roleVariable in zustand
        const setRoleVariable = useStore((state) => state.setRoleVariable);
        //inc={inc}
        return <Login navigation={navigation} setRoleVariable={setRoleVariable} set_UserName={set_UserName}></Login>;
    };

    const CreateNewBooklet = ({ navigation, route }) => {
        return <CreateNewBookletPage navigation={navigation} route={route}></CreateNewBookletPage>;
    };

    const Class = ({ navigation }) => {
        return <ClassFrame navigation={navigation}></ClassFrame>;
    };

    const StudentDemo = ({ navigation }) => {
        return <DemoBooklet navigation={navigation}></DemoBooklet>;
    };

    // main func return
    return (
        <NavigationContainer>
            {/*<Text> headerShown:false</Text>*/}
            <Stack.Navigator screenOptions={{ headerMode: false }}>
                <Stack.Screen name="LoginName" component={LoginPage} />
                <Stack.Screen
                    name="HomeName"
                    component={NewCommonBar}
                    options={({ navigation, route }) => ({
                        headerTitle: (props) => <Bar {...props} />,
                    })}
                />
                <Stack.Screen
                    name="SignUpName"
                    component={SignUp}
                    options={({ navigation, route }) => ({
                        headerTitle: (props) => <LogoTitle1 {...props} />,
                    })}
                />
                <Stack.Screen
                    name="SettingName"
                    component={Setting}
                    options={({ navigation, route }) => ({
                        headerTitle: (props) => <LogoTitle1 {...props} />,
                    })}
                />
                <Stack.Screen
                    name="ForgotPasswordName"
                    component={ForgetPassword}
                    options={({ navigation, route }) => ({
                        headerTitle: (props) => <LogoTitle2 {...props} />,
                    })}
                />
                <Stack.Screen
                    name="CreateNewBookletPage"
                    component={CreateNewBooklet}
                    options={({ navigation, route }) => ({
                        headerTitle: (props) => <Text>CreateNewBookletPage</Text>,
                    })}
                />
                <Stack.Screen
                    name="TeacherStudentConversation"
                    component={TeacherStudentConversation}
                    options={({ navigation, route }) => ({
                        headerTitle: (props) => <Text>TeacherStudentConversation</Text>,
                    })}
                />
                <Stack.Screen
                    name="Class"
                    component={Class}
                    options={({ navigation, route }) => ({
                        headerTitle: (props) => <Text>Class</Text>,
                    })}
                />
                <Stack.Screen
                    name="StudentDemo"
                    component={StudentDemo}
                    options={({ navigation, route }) => ({
                        headerTitle: (props) => <Text>StudentDemo</Text>,
                    })}
                />
            </Stack.Navigator>
        </NavigationContainer>
    );
}
