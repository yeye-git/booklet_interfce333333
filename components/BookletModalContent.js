import { useState } from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';
import { AutoGrowingTextInput } from 'react-native-autogrow-textinput';
import UploadImage from './UploadImage';

export const SingleQuestion = ({ value = {}, onChange }) => {
    const { question, answer } = value;
    const [qsValue, setQsValue] = useState({ ...value });

    const handleInputChange = (key, v) => {
        const tmp = {
            ...qsValue,
            [key]: v,
        };
        setQsValue(tmp);
        onChange(tmp);
    };

    return (
        <View style={styles.wrapper}>
            <Text style={styles.title}>Short Response</Text>
            <Text style={styles.headText}>Question:</Text>
            <AutoGrowingTextInput
                style={styles.input}
                placeholder={'Enter Your Question'}
                placeholderTextColor={'black'}
                keyboardType={'default'}
                value={qsValue.question}
                onChangeText={(t) => handleInputChange('question', t)}
            />
            <Text style={styles.headText}>Answer:</Text>
            <AutoGrowingTextInput
                // editable={false}
                style={styles.input}
                placeholder={'Enter Your Answer'}
                placeholderTextColor={'black'}
                keyboardType={'default'}
                value={qsValue.answer}
                onChangeText={(t) => handleInputChange('answer', t)}
            />
        </View>
    );
};

export const MultiSelectQuestion = ({ question = '', options = [], onChange, onOptionChange }) => {
    const [value, setValue] = useState(question);
    const [localOptions, setLocalOptions] = useState(options);

    const handleQuestionChange = (v) => {
        setValue(v);
        onChange(v);
    };

    const handleOptionChange = (i, v, key) => {
        const tmp = [...localOptions];
        tmp[i][key] = v;

        setLocalOptions(tmp);
        onOptionChange(tmp);
    };
    return (
        <View style={styles.wrapper}>
            <Text style={styles.title}>Multiple Choice</Text>
            <Text style={styles.headText}>Question:</Text>
            <AutoGrowingTextInput
                // editable={false}
                style={styles.input}
                placeholder={'Enter Your Answer'}
                placeholderTextColor={'black'}
                keyboardType={'default'}
                value={value}
                onChangeText={handleQuestionChange}
            />
            {localOptions.map((o, i) => (
                <View key={i} style={styles.row}>
                    <TouchableOpacity onPress={() => handleOptionChange(i, !o.checked, 'checked')}>
                        <View style={{ ...styles.circle, backgroundColor: o.checked ? '#5141B6' : '#fff' }} />
                    </TouchableOpacity>

                    <AutoGrowingTextInput
                        placeholder={'Answer A'}
                        placeholderTextColor={'black'}
                        onChangeText={(text) => {
                            handleOptionChange(i, text, 'label');
                        }}
                        value={o.label}
                        style={styles.textInput}
                        keyboardType={'default'}
                    />
                </View>
            ))}
        </View>
    );
};

export const ImageQuestion = ({ value = {}, onChange }) => {
    const { question, answer } = value;
    const [qsValue, setQsValue] = useState({ ...value });

    const handleInputChange = (key, v) => {
        const tmp = {
            ...qsValue,
            [key]: v,
        };
        setQsValue(tmp);
        onChange(tmp);
    };

    return (
        <View style={styles.wrapper}>
            <Text style={styles.title}>Image</Text>
            <UploadImage />
            <AutoGrowingTextInput
                style={styles.input}
                placeholder={'Enter Your Question'}
                placeholderTextColor={'black'}
                keyboardType={'default'}
                value={qsValue.question}
                onChangeText={(t) => handleInputChange('question', t)}
            />
            <AutoGrowingTextInput
                // editable={false}
                style={styles.input}
                placeholder={'Enter Your Answer'}
                placeholderTextColor={'black'}
                keyboardType={'default'}
                value={qsValue.answer}
                onChangeText={(t) => handleInputChange('answer', t)}
            />
        </View>
    );
};

export const DeleteContent = () => {
    return (
        <View style={styles.wrapper}>
            <Text style={styles.title}>Delete</Text>
            <Text style={styles.headText}>Do you want to delete this booklet?</Text>
        </View>
    );
};

export const SaveContent = () => {
    return (
        <View style={styles.wrapper}>
            <Text style={styles.title}>Save</Text>
            <Text style={styles.headText}>Would you like to save your booklet?</Text>
        </View>
    );
};

export const PublishContent = () => {
    return (
        <View style={styles.wrapper}>
            <Text style={styles.title}>Publish</Text>
            <Text style={styles.headText}>Do you want to publish the booklet?</Text>
        </View>
    );
};

const styles = StyleSheet.create({
    wrapper: {
        width: '100%',
        display: 'flex',
        flexDirection: 'column',
    },
    title: {
        fontSize: 28,
        fontWeight: 'bold',
        textAlign: 'center',
        margin: 10,
    },
    headText: {
        fontSize: 20,
        fontWeight: 'bold',
        textAlign: 'center',
    },
    input: {
        padding: 10,
        margin: 5,
        borderWidth: 3,
        borderRadius: 15,
        borderColor: 'purple',
        backgroundColor: 'white',
        minHeight: 60,
        maxHeight: 60,
        width: '100%',
        alignItems: 'center',
    },
    row: {
        width: '70%',
        // margin: '0px auto',
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'flex-start',
        alignItems: 'center',
        height: 50,
    },
    check: {
        flex: 3,
        marginTop: 20,
    },
    circle: {
        width: 26,
        height: 26,
        borderRadius: '50%',
        border: '1px solid purple',
    },
    textInput: {
        padding: 10,
        margin: 5,
        borderWidth: 1,
        borderRadius: 15,
        borderColor: 'purple',
        backgroundColor: 'white',
        height: 40,
        width: '100%',
        marginLeft: 20,
    },
});
