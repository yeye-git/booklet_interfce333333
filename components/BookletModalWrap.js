import { Modal, View, TouchableOpacity, Text, StyleSheet } from 'react-native';

export const BookletModalWrap = ({ open, children, onOk, onCancel, ...props }) => {
    return (
        <Modal visible={open} {...props} style={styles.container}>
            <View style={styles.wrapper}>
                <View style={styles.main}>
                    <View style={styles.content}>{children}</View>
                    <View style={styles.footer}>
                        <TouchableOpacity style={styles.btn} onPress={onOk}>
                            <Text style={styles.btnText}>Ok</Text>
                        </TouchableOpacity>
                        <TouchableOpacity style={styles.btn} onPress={onCancel}>
                            <Text style={styles.btnText}>Cancel</Text>
                        </TouchableOpacity>
                    </View>
                </View>
            </View>
        </Modal>
    );
};

const styles = StyleSheet.create({
    container: {
        width: '100%',
        height: '100%',
    },
    wrapper: {
        width: '100%',
        height: '100%',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
    },
    main: {
        width: '95%',
        backgroundColor: '#ADD8E6',
        borderRadius: 20,
        padding: 35,
        shadowColor: '#000',
        shadowOpacity: '0.25',
        shadowRadius: 4,
        elevation: 5,
    },
    content: {
        width: '100%',
        height: 'auto',
        backgroundColor: '#f6cefc',
        padding: 20,
    },
    footer: {
        width: '100%',
        height: 100,
        backgroundColor: '#DCBDCB',
        display: 'flex',
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
    },
    btn: {
        backgroundColor: '#ECE5FF',
        padding: 5,
        borderRadius: 10,
        width: '30%',
        height: '90%',
        display: 'flex',
        alignContent: 'center',
        justifyContent: 'center',
        marginLeft: 30,
        marginRight: 30,
    },
    btnText: {
        fontSize: 16,
        fontWeight: 'bold',
        textAlign: 'center',
    },
});
