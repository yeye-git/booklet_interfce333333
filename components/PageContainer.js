import React, { useEffect } from 'react';
import { View, Text, StyleSheet } from 'react-native';

const PageContainer = ({ children }) => {
    return <View>{children}</View>;
};

export default PageContainer;

const styles = StyleSheet.create({
    container: {
        width: '100%',
        height: '100%',
        backgroundColor: '#fff',
    },
});
