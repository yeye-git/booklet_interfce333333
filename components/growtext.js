import * as React from 'react';
import { AutoGrowingTextInput } from 'react-native-autogrow-textinput';
import {
    StyleSheet,
    Text,
    View,
    Button,
    Image,
    TouchableOpacity,
    TouchableWithoutFeedback,
    Keyboard,
} from 'react-native';
import { MCQButtons } from './styles';
import { useState, useEffect } from 'react';
import * as ImagePicker from 'expo-image-picker';

export var Q1 = '';
export var A1 = '';
export var D1 = '';
export var BD1 = '';
export var MCQuestion = '';
export var MC1 = '';
export var MC2 = '';
export var MC3 = '';
export var MC4 = '';
export var ImgQ = '';
export var ImgA = '';
export var subduedate = '';
export var qcolor = [false, false, false, false];

export function clearColor() {
    qcolor = [false, false, false, false];
}

export const GrowTextQuestion = (props) => {
    const [Question, onChangeQuestion] = useState(null);
    const { onChange, value } = props;
    React.useEffect(() => {
        onChangeQuestion(value);
    }, [value]);
    return (
        <AutoGrowingTextInput
            placeholder={'Enter Your Question'}
            placeholderTextColor={'black'}
            onChangeText={(text) => {
                onChangeQuestion(text);
                Q1 = text;
                console.log('change', text, onChange);
                if (onChange) {
                    onChange(text);
                }
            }}
            value={Question}
            style={styles.textInput}
            keyboardType={'default'}
        ></AutoGrowingTextInput>
    );
};

export const GrowTextAnswer = (props) => {
    const [Answer, onChangeAnswer] = useState(null);
    const { onChange, value } = props;
    React.useEffect(() => {
        onChangeAnswer(value);
    }, [value]);
    return (
        <AutoGrowingTextInput
            placeholder={'Enter Your Answer'}
            placeholderTextColor={'black'}
            onChangeText={(text) => {
                onChangeAnswer(text);
                A1 = text;
                if (onChange) {
                    onChange(text);
                }
            }}
            value={Answer}
            style={styles.textInput}
            keyboardType={'default'}
        ></AutoGrowingTextInput>
    );
};

export const GrowTextDisplay = () => {
    const [DisAnswer, onChangeDisAnswer] = useState(null);
    return (
        <AutoGrowingTextInput
            placeholderTextColor={'black'}
            onChangeText={(text) => {
                onChangeDisAnswer(text);
            }}
            value={DisAnswer}
            style={styles.textInput}
            keyboardType={'default'}
        ></AutoGrowingTextInput>
    );
};

export const GrowTextMcQuestion = () => {
    const [McQuestion, onChangeMcQuestion] = useState(null);
    return (
        <AutoGrowingTextInput
            placeholder={'Enter Your Question'}
            placeholderTextColor={'black'}
            onChangeText={(text) => {
                onChangeMcQuestion(text);
                MCQuestion = text;
            }}
            value={McQuestion}
            style={styles.textInput}
            keyboardType={'default'}
        ></AutoGrowingTextInput>
    );
};

export const GrowTextMC1 = () => {
    const [Answer, onChangeAnswer] = useState(null);
    return (
        <AutoGrowingTextInput
            placeholder={'Answer A'}
            placeholderTextColor={'black'}
            onChangeText={(text) => {
                onChangeAnswer(text);
                MC1 = text;
            }}
            value={Answer}
            style={styles.textInput}
            keyboardType={'default'}
        ></AutoGrowingTextInput>
    );
};

export const GrowTextMC2 = () => {
    const [Answer, onChangeAnswer] = useState(null);
    return (
        <AutoGrowingTextInput
            placeholder={'Answer B'}
            placeholderTextColor={'black'}
            onChangeText={(text) => {
                onChangeAnswer(text);
                MC2 = text;
            }}
            value={Answer}
            style={styles.textInput}
            keyboardType={'default'}
        ></AutoGrowingTextInput>
    );
};

export const GrowTextMC3 = () => {
    const [Answer, onChangeAnswer] = useState(null);
    return (
        <AutoGrowingTextInput
            placeholder={'Answer C'}
            placeholderTextColor={'black'}
            onChangeText={(text) => {
                onChangeAnswer(text);
                MC3 = text;
            }}
            value={Answer}
            style={styles.textInput}
            keyboardType={'default'}
        ></AutoGrowingTextInput>
    );
};

export const GrowTextMC4 = () => {
    const [Answer, onChangeAnswer] = useState(null);
    return (
        <AutoGrowingTextInput
            placeholder={'Answer D'}
            placeholderTextColor={'black'}
            onChangeText={(text) => {
                onChangeAnswer(text);
                MC4 = text;
            }}
            value={Answer}
            style={styles.textInput}
            keyboardType={'default'}
        ></AutoGrowingTextInput>
    );
};
export function setColors(qnum, selection) {
    if (selection == false) {
        // green
        qcolor[qnum] = true;
    } else {
        // trans
        qcolor[qnum] = false;
    }

    console.log(qcolor);
}

export const OptionMC1 = () => {
    const [selected, setSelected] = useState(false);
    return (
        <View style={styles.optionView}>
            <TouchableOpacity
                onPress={() => {
                    setSelected(!selected);
                    setColors(0, selected);
                }}
                style={{ backgroundColor: selected ? 'green' : 'transparent' }}
            >
                <MCQButtons />
            </TouchableOpacity>
        </View>
    );
};
export const OptionMC2 = () => {
    const [selected, setSelected] = useState(false);
    return (
        <View style={styles.optionView}>
            <TouchableOpacity
                onPress={() => {
                    setSelected(!selected);
                    setColors(1, selected);
                }}
                style={{ backgroundColor: selected ? 'green' : 'transparent' }}
            >
                <MCQButtons />
            </TouchableOpacity>
        </View>
    );
};
export const OptionMC3 = () => {
    const [selected, setSelected] = useState(false);
    return (
        <View style={styles.optionView}>
            <TouchableOpacity
                onPress={() => {
                    setSelected(!selected);
                    setColors(2, selected);
                }}
                style={{ backgroundColor: selected ? 'green' : 'transparent' }}
            >
                <MCQButtons />
            </TouchableOpacity>
        </View>
    );
};
export const OptionMC4 = () => {
    const [selected, setSelected] = useState(false);
    return (
        <View style={styles.optionView}>
            <TouchableOpacity
                onPress={() => {
                    setSelected(!selected);
                    setColors(3, selected);
                }}
                style={{ backgroundColor: selected ? 'green' : 'transparent' }}
            >
                <MCQButtons />
            </TouchableOpacity>
        </View>
    );
};

export const GrowTextImageQuestion = () => {
    const [ImageQuestion, onChangeImageQuestion] = useState(null);
    return (
        <AutoGrowingTextInput
            placeholder={'Enter Your Question'}
            placeholderTextColor={'black'}
            onChangeText={(text) => {
                onChangeImageQuestion(text);
                ImgQ = text;
            }}
            value={ImageQuestion}
            style={styles.textInput}
            keyboardType={'default'}
        ></AutoGrowingTextInput>
    );
};

export const GrowTextImageAnswer = () => {
    const [ImageAnswer, onChangeImageAnswer] = useState(null);
    return (
        <AutoGrowingTextInput
            placeholder={'Enter Your Answer'}
            placeholderTextColor={'black'}
            onChangeText={(text) => {
                onChangeImageAnswer(text);
                ImgA = text;
            }}
            value={ImageAnswer}
            style={styles.textInput}
            keyboardType={'default'}
        ></AutoGrowingTextInput>
    );
};

export const GrowTextDescription = () => {
    const [Description, onChangeDescription] = useState(null);
    return (
        <AutoGrowingTextInput
            placeholder={'Enter your image description'}
            placeholderTextColor={'black'}
            onChangeText={(text) => {
                onChangeDescription(text);
                D1 = text;
            }}
            value={Description}
            style={styles.textInput}
            keyboardType={'default'}
        ></AutoGrowingTextInput>
    );
};
export const GrowTextDate = () => {
    const [Date, onChangeDate] = useState(null);
    return (
        <AutoGrowingTextInput
            placeholder={'YYYY-MM-DD HH-mm-ss'}
            onChangeText={(text) => {
                onChangeDate(text);
                subduedate = text;
            }}
            value={Date}
            style={styles.textInput}
            keyboardType={'numeric'}
        ></AutoGrowingTextInput>
    );
};
export const GrowTextBookletDescription = (props) => {
    const [BDescription, onChangeBookDescription] = useState(null);
    const { onChange } = props;
    return (
        <AutoGrowingTextInput
            placeholder={'Enter your booklet description'}
            placeholderTextColor={'black'}
            onChangeText={(text) => {
                onChangeBookDescription(text);
                BD1 = text;
                if (onChange) {
                    onChange(text);
                }
            }}
            value={BDescription}
            style={styles.textInput}
            keyboardType={'default'}
        ></AutoGrowingTextInput>
    );
};

const styles = StyleSheet.create({
    textInput: {
        padding: 10,
        margin: 5,
        borderWidth: 3,
        borderRadius: 15,
        borderColor: 'purple',
        backgroundColor: 'white',
        minHeight: 60,
        maxHeight: 60,
        width: '100%',
        alignItems: 'center',
    },
    optionView: {
        width: '30%',
        alignItems: 'center',
        marginTop: '6%',
        margin: 1,
    },
});
export function clearDate() {
    subduedate = '';
}
