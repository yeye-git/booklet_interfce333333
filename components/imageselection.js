import * as React from 'react';
import { Text, View, Button, Image, TouchableOpacity} from 'react-native';
import { useState, useEffect } from 'react';
import * as ImagePicker from 'expo-image-picker';
import { NoAccessToCameraTextCont, NoAccessToCameraText, HaveAccessToCameraCont } from './styles';
//import Video from 'react-native-video';
export var chosenimage = null;
export default function SelectImage() {
    const [askGalleryPermission, setAskGalleryPermission] = useState(null);
    const [image, setImage] = useState(null);
    //const [video, setVideo] = useState(null);

    useEffect(() => {
        (async () => {
    const galleryStatus = await ImagePicker.requestMediaLibraryPermissionsAsync();
          setAskGalleryPermission(galleryStatus.status === 'granted');
    })();
    }, []);

    const pickImage = async () => {
        let result = await ImagePicker.launchImageLibraryAsync({
          mediaTypes: ImagePicker.MediaTypeOptions.allowsEditing,
          allowsEditing: true,
          aspect: [1, 1],
          quality: 1,
        });
    console.log(result);

    if (!result.cancelled) {
        setImage(result.uri);
      }
    };

    if (askGalleryPermission === false) {
        return (
            <NoAccessToCameraTextCont>
                <NoAccessToCameraText>No Access to Photo Gallary.</NoAccessToCameraText>
            </NoAccessToCameraTextCont>
        );

    }
    else{
        clearMyImage();
        return (
            <HaveAccessToCameraCont>
                
                <TouchableOpacity style={{ margin: '1%', backgroundColor: 'skyblue' }}>
                    <Button color="purple"  title="Click To Open Photo Gallery" onPress={() => pickImage()} />
                </TouchableOpacity>
                {image && <Image source={{uri: image}} style={{ flex:1/1, margin: '1%'}}/>}
                {setMyImage(image)}
            </HaveAccessToCameraCont>
        );
    }
}
export function setMyImage(myimage){
    chosenimage = myimage;
}
export function clearMyImage(){
    chosenimage = null;
}


