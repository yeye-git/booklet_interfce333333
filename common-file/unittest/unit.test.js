import CreatNewBookletPage, { noofques } from '../../screens/teacher/CreateNewBookletPage';
import {questions} from '../../screens/teacher/CreateNewBookletPage';
import {sum} from './sum';
import React from 'react'
import renderer from 'react-test-renderer';
import App from '../../App'
import { GrowTextAnswer } from '../../components/growtext';
describe('sum', () => {   
    test('should sum 2 nums', () => {
        expect(sum(1,2)).toBe(3);
    });
});

describe('Ensure no of questions is empty', () => {   
    test('questions array should be fuller', () => {
        
        expect(noofques).toBe(0);
    });
});
describe('Ensure no of questions array is empty', () => {   
    test('questions array should be fuller', () => {
        
        expect(questions).toStrictEqual([]);
    });
});


