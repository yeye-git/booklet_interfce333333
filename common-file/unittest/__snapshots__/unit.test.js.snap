// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  className="sc-fIavCj jdbtbh"
>
  <Modal
    animationType="slide"
    className="sc-dWINGa hIYWEI"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  />
  <View
    className="sc-duzrYq djdzQo"
  />
  <View
    className="sc-dkdnUF hreBFH"
  >
    <View
      className="sc-hAsxaJ iMuqYO"
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          className="sc-jqUVSM exFXRc"
          resizeMode="cover"
          source={1}
        />
      </View>
    </View>
    <TextInput
      className="sc-hjriPb evmmLA"
      keyboardType="default"
      onChangeText={[Function]}
      placeholder="Enter Booklet Heading"
      value={null}
    />
  </View>
  <View
    className="sc-WCkqM bQngZE"
  >
    <View
      className="sc-ikjQzJ bSPrNs"
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          className="sc-jqUVSM exFXRc"
          resizeMode="cover"
          source={1}
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          className="sc-jqUVSM exFXRc"
          resizeMode="cover"
          source={1}
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          className="sc-jqUVSM exFXRc"
          resizeMode="cover"
          source={1}
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          className="sc-jqUVSM exFXRc"
          resizeMode="cover"
          source={1}
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          className="sc-jqUVSM exFXRc"
          resizeMode="cover"
          source={1}
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          className="sc-jqUVSM exFXRc"
          resizeMode="cover"
          source={1}
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          className="sc-jqUVSM exFXRc"
          resizeMode="cover"
          source={1}
        />
      </View>
    </View>
    <RCTScrollView
      className="sc-tsFYE jEFiMj"
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
